tests/installation/first_boot.pm

```diff
@@ -102,24 +103,7 @@ sub test_flags {
 }
 
 sub post_fail_hook {
-    my $self = shift;
-
-    # Reveal what is behind Plymouth splash screen
-    wait_screen_change {
-        send_key 'esc';
-    };
-    # save a screenshot before trying further measures which might fail
-    save_screenshot;
-    # if we found a shell, we do not need the memory dump
-    if (!(match_has_tag('emergency-shell') or match_has_tag('emergency-mode'))) {
-        die "save_memory_dump is temporarily unavailable, see https://progress.opensuse.org/issues/19390";
-        die "save_memory_dump not implemented, no way to save memory_dump" unless check_var('BACKEND', 'qemu');
-        diag 'Save memory dump to debug bootup problems, e.g. for bsc#1005313';
-        save_memory_dump;
-    }
-
-    # try to save logs as a last resort
-    $self->export_logs();
+    bootbasetest->post_fail_hook();
```

Comment:
you might want to export the function, although I didn't know that calling directly with `->` could work.

Reply:
Thanks for the hint, I was wondering like an Idiot why it did not work as expected :D

Reply:
Hmm after taking a look at y2logstep and opensusebasetest I wonder if it is needed to export the function, none of those baseclasses do it?

Reply:
for example [here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/lvm_thin_check.pm#L69) is calling the parent because is overriding the function in `base` but you use it as utility, not as base. As a base you can also call it sometime with `$self`, it depends on the case.
As a utility library I would say that you can use [this approach](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/yast2_shortcuts.pm#L21)

Reply:
@Martchus wdyt?

Reply:
I wouldn't export it. That makes accidentally calls to that function more likely.

Not sure why the `::` syntax doesn't work but `->` should be fine, too (since the function does not evaluate any arguments anyways).

Reply:
As an alternative I think you could use multiple inheritance, e.g. `use base qw(foo bar)`

Reply:
@okurz I saw the same alternative, and would actually create a ticket or suggest @DrMullings to come with this approach, but the pr has been hanging here for so long... that it started to get sad and feel forgotten...

Reply:
As I already told @foursixnine multiple inheritance sounds like opening the gates to the hell of ambiguity, we should not do it as long as we have a working state